### DISCUSSION TITLE
List all my Github Pages Projects

### DISCUSSION THREAD

**Discussion by quentinhnrt** [community/community#24595] [Apps, API and Webhooks]
<p>Hi !</p>
<p>I’m trying to make a list of all my Github Pages Projects (<a href="http://username.github.io/project-name" rel="noopener nofollow ugc">username.github.io/project-name</a>) but I can’t find a way to do it with a request on the Github Api. Like I can get all my repos but not only the ones that are hosted on Github Pages</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>try</p>
<p><a href="https://docs.github.com/en/rest/repos/repos#list-repositories-for-the-authenticated-user">https://docs.github.com/en/rest/repos/repos#list-repositories-for-the-authenticated-user</a></p>
<p>then look at</p>
<p><code>deployments_url</code></p>
<p>there you’ll know whether it’s hosted on GitHub Pages, or any other deployments</p>

**Comment by domfarolino** [comment]
I created ~https://gh-pages.glitch.me/~ https://gh-pages-vato.onrender.com to help you browse through your repositories that have GitHub pages enabled as well as those for the people you follow:

<img width="853" alt="Screenshot 2023-05-03 at 7 46 02 AM" src="https://user-images.githubusercontent.com/9669289/235917563-d30781da-3086-4170-ae7c-b0ce78de6b56.png">

**Reply by bukarinevg** [reply]
great project

**Reply by Kreijstal** [reply]
is dead

**Reply by guibranco** [reply]
It is now here: https://gh-pages-vato.onrender.com/ and the repository is here: https://github.com/domfarolino/pages

**Comment by lucafaggianelli** [comment]
Hey I had the same issue and I came up with this shell one liner:

1. Be sure to have installer the `jq` cli
2. Put your username

> You don't need any API key, the API call will fetch only public repos

```sh
curl "https://api.github.com/users/<username>/repos?per_page=100" | jq '.[] | select(.has_pages==true) | .name'
```

**Comment by ghost** [comment]
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2026-03-10" \
  https://api.github.com/
